sway-lib-std/src/storage.sw

```diff
@@ -681,6 +681,262 @@ impl<V> StorageVec<V> {
+    ///     storage.vec.push(15);
+    ///     storage.vec.reverse();
+    ///
+    ///     assert(15 == storage.vec.get(0).unwrap());
+    ///     assert(10 == storage.vec.get(1).unwrap());
+    ///     assert(5 == storage.vec.get(2).unwrap());
+    /// }
+    /// ```
+    #[storage(read, write)]
+    pub fn reverse(self) {
+        let len = get::<u64>(__get_storage_key()).unwrap_or(0);
+
+        if len < 2 {
+            return;
+        }
+
+        let mid = len / 2;
+        let mut i = 0;
+        while i < mid {
+            let element1_key = sha256((i, __get_storage_key()));
+            let element2_key = sha256((len - i - 1, __get_storage_key()));
+
+            let element1_value: V = get::<V>(element1_key).unwrap();
+            store::<V>(element1_key, get::<V>(element2_key).unwrap());
+            store::<V>(element2_key, element1_value);
+
+            i += 1;
+        }
+    }
+
+    /// Fills `self` with elements by cloning `value`.
+    ///
+    /// ### Arguments
+    ///
+    /// * value - Value to copy to each element of the vector.
+    ///
+    /// ### Storage Access
+    ///
+    /// * Reads: `1`
+    /// * Writes: `self.len()`
+    ///
+    /// ### Examples
+    ///
+    /// ```sway
+    /// storage {
+    ///     vec: StorageVec<u64> = StorageVec {},
+    /// }
+    ///
+    /// fn foo() {
+    ///     storage.vec.push(5);
+    ///     storage.vec.push(10);
+    ///     storage.vec.push(15);
+    ///     storage.vec.fill(20);
+    ///
+    ///     assert(20 == storage.vec.get(0).unwrap());
+    ///     assert(20 == storage.vec.get(1).unwrap());
+    ///     assert(20 == storage.vec.get(2).unwrap());
+    /// }
+    /// ```
+    #[storage(read, write)]
+    pub fn fill(self, value: V) {
+        let len = get::<u64>(__get_storage_key()).unwrap_or(0);
+
+        let mut i = 0;
+        while i < len {
+            store::<V>(sha256((i, __get_storage_key())), value);
+            i += 1;
+        }
+    }
+
+    /// Resizes `self` in place so that `len` is equal to `new_len`.
+    ///
+    /// If `new_len` is greater than `len`, `self` is extended by the difference, with each
+    /// additional slot being filled with `value`. If the `new_len` is less than `len`, `self` is
+    /// simply truncated.
+    ///
+    /// ### Arguments
+    ///
+    /// new_len - The new length to expand or truncate to
+    /// value - The value to fill into new slots if the `new_len` is greater than the current length
+    ///
+    /// ### Storage Access
+    ///
+    /// * Reads - `1`
+    /// * Writes - `if new_len > self.len() { new_len - len + 1 } else { 1 }`
+    ///
+    /// ### Examples
+    ///
+    /// ```sway
+    /// storage {
+    ///     vec: StorageVec<u64> = StorageVec {},
+    ///     vec2: StorageVec<u64> = StorageVec {},
+    /// }
+    ///
+    /// fn foo() {
+    ///     storage.vec.push(5);
+    ///     storage.vec.push(10);
+    ///     storage.vec.push(15);
+    ///     storage.vec.resize(4, 20);
```

Comment:
Pre/post of length and capacity of a single vector is fine. No need for two. I would also resize it to a greater size than 1 e.g. 2 or 3. This demonstrates that additional slot(s) are filled and not just 1 otherwise I need to check the code to ensure that is the case.